source/s3_meta_request.c

```diff
@@ -115,17 +120,27 @@ static void s_s3_request_on_body(
     struct s3_meta_request_binding *request_binding = user_data;
     if (request_binding->file) {
         fwrite((void *)body->ptr, body->len, 1, request_binding->file);
-        return;
     }
 
     /*************** GIL ACQUIRE ***************/
     PyGILState_STATE state;
     if (aws_py_gilstate_ensure(&state)) {
         return; /* Python has shut down. Nothing matters anymore, but don't crash */
     }
-    PyObject *result = PyObject_CallMethod(
-        request_binding->self_py, "_on_body", "(y#K)", (const char *)(body->ptr), (Py_ssize_t)body->len, range_start);
+    PyObject *result;
+    if (request_binding->file) {
```

Comment:
these should likely be batched and sent something like once per second, instead of on each callback and be done outside of the context of the GIL